src/ray/core_worker/actor_handle.cc

```diff
@@ -58,10 +58,26 @@ void ActorHandle::SetActorTaskSpec(TaskSpecBuilder &builder,
 
 void ActorHandle::Serialize(std::string *output) { inner_.SerializeToString(output); }
 
-void ActorHandle::Reset() {
-  std::unique_lock<std::mutex> guard(mutex_);
-  task_counter_ = 0;
-  actor_cursor_ = ObjectID::FromBinary(inner_.actor_cursor());
+void ActorHandle::UpdateLocation(const ClientID &node_id) {
+  state_ = gcs::ActorTableData::ALIVE;
+  node_id_ = node_id;
+}
+
+void ActorHandle::MarkFailed(bool reset_task_counter) {
+  state_ = gcs::ActorTableData::RECONSTRUCTING;
+  node_id_ = ClientID::Nil();
+  if (reset_task_counter) {
+    std::unique_lock<std::mutex> guard(mutex_);
+    task_counter_ = 0;
+    actor_cursor_ = ObjectID::FromBinary(inner_.actor_cursor());
+  }
+}
+
+void ActorHandle::MarkDead() {
+  state_ = gcs::ActorTableData::DEAD;
+  RAY_CHECK(state_.has_value());
```

Comment:
Oh oops, this was a debugging thing, I'll remove it.

I agree, but we'll need to do something different for these sorts of errors. There is no client code that can catch this error because it's called in an async handler.